### ISSUE TITLE
About the reproducibility and validation set

### ISSUE THREAD

**Issue by psm1206** [HKUDS/LightGCL#9]
Hi. I am a reader who read your paper with great interest. After running your code myself, I have two questions. I would be grateful if you could answer them when you have time.

1. When I run the command "python main.py --data yelp", the performance is higher than the performance in your paper. I'm wondering why the results are different.
**[Performance table of my experiments]**
<html>
<body>
<google-sheets-html-origin>

  |   | Recall@20 | Recall@40 | NDCG@20 | NDCG@40
-- | -- | -- | -- | -- | --
Yelp | reported | 0.0793 | 0.1292 | 0.0668 | 0.0778
  | reproduced | 0.1001 | 0.1587 | 0.0868 | 0.1080
Gowalla | reported | 0.1578 | 0.2245 | 0.0935 | 0.1108
  | reproduced | 0.2124 | 0.2993 | 0.1236 | 0.1464

</body>
</html>

2. In the paper, the validation set is used 20% of the whole data, but in the code, it is not used. Also, early stopping is not used in the code, so I'm wondering if the reported results in the paper are not using early stopping.

Thank you for your time.
I look forward to hearing back from you.

**Comment by Rick-Cai** [comment]
Hi, thank you for reading our paper.

1. The reported results and the results produced by the released codes are different because after we submitted our paper, we did more experiments and found that there is a better experimental setting that can produce a higher performance, so we updated the codes in this repo to the new setting. The differences between the old and new settings are mainly on the sampling of the postive and negative items, while the model structure remains unchanged.

2. We tuned our model with the validation set to obtain the optimal hyperparameters and results reported in the paper. The validation set is for hyperparameter tuning only, and the released codes do not involve tuning, so the validation set is not used here. For early stopping, we did not notice significant overfitting in our model training, so there is no need to adopt early stopping strategies.

Please feel free to contact us if there is still unclarity.

**Comment by psm1206** [comment]
Thank you for your quick and kind response.

Additionally, I have two questions about question 1.

1-1. Did you observe a performance improvement in other baselines (e.g., HCCF, SHT, SimGCL) like that in LightGCL?

1-2. The hyperparameter lambda1 is set to 0.2 in the code unlike the range {1e-5, 1e-6, 1e-7} presented in the paper, did this have a significant impact on the performance improvement?

Thanks again for your kind response.

**Comment by Rick-Cai** [comment]
1. Yes, the change in setting will improve the performance of most cl-based methods, but the relative performance between models is relatively stable. We are still conducting related experiments in the new setting. Once there are new results, we will update them to arXiv.
2. After the setting is changed, the hyperarameter tuning needs to be redone. The change in sampling method will directly affect the scale of the gradients in each step. Thus, the originally proposed range is no longer applicable. Not only the numbers need to change, but as you see the scale of the parameters is also changed to suit the new setting.

**Comment by psm1206** [comment]
I totally understood the questions I asked thanks to you.

Lastly, if possible, could you tell me about the sampling method before and after the change?

Thank you so much for your kind response.

**Comment by Rick-Cai** [comment]
The old setting is to sample items by user, i.e., for each user, we sample a fixed number of positive items (from this user's neighbor) and negative items.
The new setting is to directly sample the interactions, i.e., directly sample (user, item) pairs from the graph. The number of samples involving each user in the new setting is uncertain.

**Comment by psm1206** [comment]
Thank you so much!
I will close this issue.

Congratulations on your paper acceptance!
